File: Scopes_and_closure/closure.py
#lets grab knowledge about closure

# In Python, functions are just objects — like strings or numbers. You can:

# Assign them to variables

# Pass them as arguments

# Return them from other functions


def func1():
    X=50
    def func2():
        
      print(X)
    return func2     # here it just return func2
result = func1()
result()

 #Another Exmaple
def chai_code(num):
     def actual(x):
         return x ** num
     return actual
f = chai_code(2)
g =chai_code(3)
print(f(3))
print(g(3))

File: Scopes_and_closure/closure.js
function func1(){
    let x = 100;
    function func2(){
       console.log(x)
    }
    return func2    //func1 only return func2
}
let result = func1() //first this function is called   
result()   //here func2 is called which is stored in result and this help to print x=100